### PR TITLE
Mdelcid week4

### PR DESCRIPTION
**Pull request by MarDelCid**

### CODE REVIEW

File: Mdelcid_wk4/guessing_game.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		else
+			print "Try Again! "
+			end
+		end
+	end
+end
+
+print "Would you like to guess a number? (y/n) "
+play = gets.chomp.downcase
+if play == "y"
```
Comment (Eldwick) [inline Mdelcid_wk4/guessing_game.rb]: Nice! If you have time, can you make this game continue to play until the user decides they are done playing? Look into `while` loops https://learnrubythehardway.org/book/ex33.html

File: Mdelcid_wk4/instacart_mini.rb [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  attr_accessor :name, :cart, :address, :orders
+
+  def initialize(name)
+    @name = name
+    @cart = []
+    @orders= []
+  end
+
+  def add_item(item)
+    @cart.push(item: item.name, price: item.price)
```
Comment (Eldwick) [inline Mdelcid_wk4/instacart_mini.rb]: I would suggest pushing in the `item` directly into the `@cart` array. That way you have access to `Item` methods later on.

```
@cart.push(item)
```

File: Mdelcid_wk4/instacart_mini.rb [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  end
+
+  def total
+    total = 0.0
+    @cart.each do |item|
```
[the remark this answers is not in the document]
Comment (Eldwick) [reply Mdelcid_wk4/instacart_mini.rb]: And if you switch to using `Item` instances within the cart, you could write this even more simply like:
```
@cart.sum(&:price)
```

File: Mdelcid_wk4/guessing_game.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+def guessing_game
+	random_number = Random.new.rand(1..100)
```
Comment (jerflan) [inline Mdelcid_wk4/guessing_game.rb]: You don't need `Random.new`. ruby will return an integer with just `rand(1..100)`

File: Mdelcid_wk4/guessing_game.rb (hunk 1 above)
Comment (jerflan) [reply Mdelcid_wk4/guessing_game.rb]: You can also look at letting the user choose the range as well.

File: Mdelcid_wk4/instacart_mini.rb [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+  puts "\n"
+
+  print_catalog
+
+  puts "\nPlease add your items to your cart! "
+  loop do
+    puts "\nWhich item would you like to add?, type 'chek out' to check out "
+    new_item = gets.chomp.downcase
+
+    case new_item
```
Comment (muffy) [inline Mdelcid_wk4/instacart_mini.rb]: You can do this much more simply like this:

```
  @new_user.add_item(@catalog[new_item..gsub(" ", "_").to_sym])
```

`gsub` will replace all spaces `" "` with underscores `"_"`
`to_sym` will convert a string, e.g. `"popsicle"` into a symbol, e.g. `:popsicle`

```
irb(main):004:0> "fruit snack".gsub(" ", "_").to_sym
=> :fruit_snack
irb(main):005:0> 
```

File: Mdelcid_wk4/instacart_mini.rb [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  else puts "You are almost there! You can check out at any time."
+  end
+
+  @new_user.orders.each do |key, value|
+    puts "#{your_shopper.name} has found the #{value}"
+  end
+
+  puts "\nYour shopper has checked out, your order is estimated to arrive on time!"
+
+  puts "\nThank you for using Instacart, we hope to see you again soon."
```
Comment (muffy) [inline Mdelcid_wk4/instacart_mini.rb]: Really nice interaction with the user.

File: Mdelcid_wk4/tiny_instacart.rb [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    puts " 🥕✨" * 10
+    puts "Good news! Your order has been placed and is being sent to a shopper. "
+    puts " 🥕✨" * 10
+    sleep (1)
+    puts "\n"
+      loading
+	puts "\nHooray! Your shopper is now shopping your oder. We will keep you updated.\n "
+	sleep(2)
+    grocery_list.each do |item|
+    	update_customer(item)
```
Comment (muffy) [inline Mdelcid_wk4/tiny_instacart.rb]: Very Instacart! 😄

File: Mdelcid_wk4/tiny_instacart.rb [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+	end
+
+end
+
+# def update_grocery_list(grocery_list, item, function)
+# 	grocery_list.delete_if { |list_item| list_item == item }
+# end
+
+def receipt(order_total)
+	puts "\n"
```
Comment (muffy) [inline Mdelcid_wk4/tiny_instacart.rb]: Your indents should be 2 spaces rather than 4.

File: Mdelcid_wk4/tiny_instacart.rb [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+	puts "$" * 20
+	puts "Your order total is $#{order_total}"
+	puts "$" * 20
+	print "\nWould you like to add a tip? (y/n)"
```
Comment (muffy) [inline Mdelcid_wk4/tiny_instacart.rb]: Very thoughtful!

File: Mdelcid_wk4/tiny_instacart.rb [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+		puts "$" * 20
+		puts "Your updated total is #{order_total + tip}"
+		puts "$" * 20
+	else puts "\nHave a great day! 🙌"
+	end
+end
+
+def loading
+	10.times do 
+    	print "  🚗💨"
```
Comment (muffy) [inline Mdelcid_wk4/tiny_instacart.rb]: 😆  cool

**Pull request closed without merging**